packages/headless/src/features/search/legacy/search-actions-thunk-processor.test.ts

```diff
@@ -248,4 +251,45 @@ describe('AsyncSearchThunkProcessor', () => {
     expect(config.extra.apiClient.search).toHaveBeenCalled();
     expect(processed.response).toMatchObject(responseAfterModification);
   });
+
+  it('does not call streaming API if there is no generative answering id', async () => {
+    const processor = new AsyncSearchThunkProcessor<{}>(config);
+
+    const searchResponse = buildMockSearchResponse({
+      results: [buildMockResult()],
+    });
+
+    const fetched = {
+      response: {
+        success: searchResponse,
+      },
+      duration: 123,
+      queryExecuted: 'foo',
+      requestExecuted: buildMockSearchRequest(),
+    };
+
+    await processor.process(fetched);
+    expect(streamAnswer).not.toHaveBeenCalled();
+  });
+
+  it('does call streaming API if there is a generative answering id', async () => {
+    const processor = new AsyncSearchThunkProcessor<{}>(config);
+
+    const searchResponse = buildMockSearchResponse({
+      extendedResults: {generativeQuestionAnsweringId: '123'},
+      results: [buildMockResult()],
+    });
+
+    const fetched = {
+      response: {
+        success: searchResponse,
+      },
+      duration: 123,
+      queryExecuted: 'foo',
+      requestExecuted: buildMockSearchRequest(),
+    };
+
+    await processor.process(fetched);
+    expect(streamAnswer).toHaveBeenCalledWith({streamId: '123'});
```

Comment:
Since you're not calling `jest.clearAllMocks()` between each test, the order of execution of those two tests is important. Not necessarily a big deal right now, but it could cause problems later on if we ever need to add / move tests that rely on this mock.